examples/pub/lib/pub_repository.dart

```diff
@@ -258,3 +265,15 @@ class PubSearchResponse with _$PubSearchResponse {
   factory PubSearchResponse.fromJson(Map<String, Object?> json) =>
       _$PubSearchResponseFromJson(json);
 }
+
+class PubProxyInterceptor extends Interceptor {
```

Comment:
A nice comment to explain why we do this would be great :)

Reply:
Right, no problem.